scripts/mkhtml.js

```diff
@@ -79,6 +81,14 @@ async function asyncGenerateHtmlContent(data) {
   const pdfTemplate = handlebars.compile(pdfFile);
   const htmlTemplate = handlebars.compile(htmlFile);
 
+  Object.keys(sortedCategories).forEach((currentKey) => {
+    const symbolsArr = sortedCategories[currentKey];
+    sortedCategories[currentKey] = {
+      anchor: slugify(currentKey),
+      symbols: symbolsArr,
+    };
+  });
+
```

Comment:
Personally I'd avoid mutating the sortedCategories and create a new one. This unmutable style can save lots of bugs and confusion when debugging. One way to do it is as in lines 70-74 above.

However, there is a subtle bug in the use of `Object.keys(sortedCategories)` here. The resulting array is officially in arbitrary order. Hence you need to call .sort() again, as in line 72. In reality, node almost certainly iterates in sequence as is being assumed, so everything works as expected. Or perhaps node (or rather V8) specifies the iteration order, I don't know.

I'm now worried about the same bug applying when the template is applied. Unless handlebars handles `#each` iteraction specially I excpect the same potenial bug applies - the [handlebars documentaion doesn't say](ttps://handlebarsjs.com/builtin_helpers.html#iteration].

Ideally, the sortedCategories and the data passed to the template would be an Array. (not sure about Map sorting).

Reply:
Oh and mutating the thing you are iterating is even more dodgy :)